asymmetric_uncertainty/core.py

```diff
@@ -82,21 +82,42 @@ def pdf(self,x):
                             [lambda x : np.sqrt(2)/np.sqrt(np.pi)/(self.plus+self.minus) * np.exp(-1*(x-self.value)**2 / (2*self.minus**2)),
                              lambda x : np.sqrt(2)/np.sqrt(np.pi)/(self.plus+self.minus) * np.exp(-1*(x-self.value)**2 / (2*self.plus**2))])
     
-    def cdf(self,x):
+    def cdf(self,x,num_sigma=5,discretization=100):
         """
         Computes and returns the values of the cumulative distribution function for the specified input.
         """
-        return np.cumsum(self.pdf(x))/np.sum(self.pdf(x))
-        
+        neg_x = np.linspace(self.value-(num_sigma*self.minus),self.value,discretization)
+        pos_x = np.linspace(self.value,self.value+(num_sigma*self.plus),discretization)
+        x_arr = np.array(list(neg_x)+list(pos_x))
+
+        if hasattr(x,'__len__'):
+            ret = np.zeros(shape=len(x))
+            for i in range(len(x)):
+                if x[i] < x_arr[0]:
+                    ret[i] = 0
+                elif x[i] > x_arr[-1]:
+                    ret[i] = 1
+                else:
+                    ret[i] = np.sum( self.pdf( x_arr[ : np.argmin(x[i] >= x_arr)] ) ) / np.sum(self.pdf(x_arr) )
+        else:
+            if x < x_arr[0]:
+                ret = 0
+            elif x > x_arr[-1]:
+                ret = 1
+            else:
+                ret = np.sum( self.pdf( x_arr[ : np.argmin(x >= x_arr)] ) ) / np.sum(self.pdf(x_arr) )
```

Comment:
Don't we expect `np.sum(self.pdf(x_arr))` to always be ~1?

Reply:
Yes, we absolutely do. Good find.

Reply:
Actually, disregard&mdash;the $\Delta x$ spacing in `x_arr` is not necessarily 1, so even though the *integral* should always be unity, the straight sum of the $y$-values might not be.

Reply:
You're right that the sum of the discretized PDF won't be 1. But the sum of the PDF times the bin size will be. 
But still, isn't the CDF just the integral of PDF from (0,x). Is the sum of the entire pdf in the denominator necessary?  
If the PDF is properly normalized (which is what you showed when you took the integral in the other thread), then the CDF should need another normalization, right?